app/instrumentation-tests/src/org/commcare/androidTests/AppInstallationTest.kt

```diff
@@ -0,0 +1,199 @@
+                .perform(click())
+
+        // Check that all the apps belong to commcare-tests domain
+        for (position in 0 until getAppListSize()) {
+            InstrumentationUtility.getSubViewInListItem(R.id.apps_list_view, position, R.id.domain)
+                    .check(matches(withText("commcare-tests")))
+        }
+
+        // Check the app names
+        matchAppInAppList("Case callout test for Simprints")
+        matchAppInAppList("Case Search and Claim")
+        matchAppInAppList("Integration Tests")
+
+        // Install 1 of the apps
+        onView(withText("Case Search and Claim"))
+                .perform(click())
+
+        assert(CommCareApplication.instance().currentApp != null, "App is null")
+        assert(CommCareApplication.instance().currentApp.appRecord.displayName == "Case Search and Claim", "App didn't match")
+    }
+
+    @Test
+    fun testAppInstall_usingWebWorkerDetails() {
+        InstrumentationUtility.openOptionsMenu()
+        onView(withText("See Apps for My User"))
+                .perform(click())
+        // Confirm the activity
+        intended(hasComponent(InstallFromListActivity::class.java.name))
+
+        // Verify that we start out in mobile user auth mode
+        onView(withId(R.id.edit_domain))
+                .check(matches(isDisplayed()))
+        onView(withId(R.id.edit_email))
+                .check(matches(not(isDisplayed())))
+
+        // Toggle switch
+        onView(withClassName(endsWith("Switch")))
+                .perform(click())
+
+        // Verify that we're now in web user auth mode
+        onView(withId(R.id.edit_domain))
+                .check(matches(not(isDisplayed())))
+        onView(withId(R.id.edit_email))
+                .check(matches(isDisplayed()))
+
+        // Test getting app list for a mobile user
+        onView(withId(R.id.edit_email))
+                .perform(typeText(BuildConfig.HQ_API_USERNAME))
+        closeSoftKeyboard()
+        onView(withId(R.id.edit_password))
+                .perform(typeText(BuildConfig.HQ_API_PASSWORD))
+        closeSoftKeyboard()
+        onView(withId(R.id.get_apps_button))
+                .perform(click())
+
+        // Check that we see each of the apps in this domain, plus the domain name
+        for (position in 0 until getAppListSize()) {
+            InstrumentationUtility.getSubViewInListItem(R.id.apps_list_view, position, R.id.domain)
+                    .check(matches(anyOf(withText("commcare-tests"), withText("swat"))))
+        }
+
+        matchAppInAppList("Case callout test for Simprints")
+        matchAppInAppList("Case Search and Claim")
+        matchAppInAppList("Demo - Form Design Patterns")
+        matchAppInAppList("Demo - Live XForm Examples")
+        matchAppInAppList("SWAT: CommCare Projects Phone Survey")
+        matchAppInAppList("SWAT: App Tracker")
+
+        // Install 1 of the apps
+        onData(allOf(`is`(instanceOf(AppAvailableToInstall::class.java)),
```

Comment:
Ohh, actually this one's a better way to click a list item, as this will automatically scroll the list if the item is not yet visible. The above one simply clicks the view in the screen, and I forgot to update the above one, though it works cuz the list isn't big. 
So I'll update the above one to use `onData` utility.